sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -2057,14 +2055,12 @@ case class ValidateExternalType(child: Expression, expected: DataType, externalD
     if (checkType(input)) {
       input
     } else {
-      throw new RuntimeException(s"${input.getClass.getName}$errMsg")
+      throw QueryExecutionErrors.invalidExternalTypeError(
+        input.getClass.getName, expected.simpleString, child)
```

Comment:
the `simpleString()` and `toSQLType` might print types differently, we should have unified representation of types in errors. Could you pass `expected` to `invalidExternalTypeError()`, and wrap it by `toSQLType` inside of it.

Reply:
`expectedType: DataType` so it can not be passed into `toSQLExpr` but I can pass it into `quoteByDefault`. It might have sense because:

```
  def toSQLExpr(e: Expression): String = {
    quoteByDefault(toPrettySQL(e))
  }
```

Reply:
`DataType` is a sub-type of `AbstractDataType`, in this way you can pass to:
```scala
def toSQLType(t: AbstractDataType): String
```